### PR TITLE
Calendar credential key missing fix

### LINKED ISSUE #4641: better error when not all google calendar

### PR DESCRIPTION
**Pull request by leog**
## What does this PR do?

For some reason we lost the warning that let the users know if there was any connected calendar without the correct permissions given.

How it looked before the fix:
<kbd></kbd>

How it looks now with the fix for one connected calendar:
<kbd></kbd>

How it looks now with the fix for two connected calendar (one failing):
<kbd></kbd>

Fixes #4641 

**Environment**: Staging(main branch) / Production

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

Go to Apps and connect a Google Calendar. When providing permission don't check the following checkboxes and click the "Continue" button.

<kbd></kbd>

<kbd></kbd>

### CODE REVIEW

File: apps/web/components/v2/apps/CalendarListContainer.tsx [hunk 1]
```diff
@@ -153,7 +153,7 @@ function ConnectedCalendarsList(props: Props) {
           return null;
         }
         return (
-          <List className="flex flex-col gap-6">
+          <List className="flex flex-col gap-6" noBorderTreatment>
```
Comment (leog) [inline apps/web/components/v2/apps/CalendarListContainer.tsx:156]: Had to make the list without borders as they were messing up each calendar integration list item, specifically the warning itself.

File: apps/web/components/v2/apps/CalendarListContainer.tsx [hunk 2]
```diff
@@ -193,10 +193,11 @@ function ConnectedCalendarsList(props: Props) {
                 ) : (
                   <Alert
                     severity="warning"
-                    title={t("calendar_error")}
-                    message={item.error?.message}
+                    title={t("something_went_wrong")}
+                    message={t("calendar_error")}
+                    iconClassName="h-10 w-10 ml-2 mr-1 mt-0.5"
```
Comment (leog) [inline apps/web/components/v2/apps/CalendarListContainer.tsx:198]: Made the icon the same size and position as the rest of the icons in the calendar integration list

File: apps/web/components/v2/apps/CalendarListContainer.tsx [hunk 3]
```diff
@@ -193,10 +193,11 @@ function ConnectedCalendarsList(props: Props) {
...
                     actions={
-                      <div className="w-32">
+                      <div className="flex w-32 justify-end">
```
Comment (leog) [inline apps/web/components/v2/apps/CalendarListContainer.tsx:200]: Disconnect icon at the very end as the other integrations

File: packages/app-store/googlecalendar/api/callback.ts [hunk 4]
```diff
@@ -36,10 +36,7 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
 
   if (code) {
     const token = await oAuth2Client.getToken(code);
-
     key = token.res?.data;
-
-    if (!key) res.status(401).json({ message: "Permissions not granted" });
```
Comment (leog) [inline packages/app-store/googlecalendar/api/callback.ts:42]: We have already covered this scenario afterwards in the workflow. This result in the API was throwing a bare bones JSON when clicking "Cancel" in Google Calendar permission flow, not a very good UX.

File: packages/core/CalendarManager.ts [hunk 5]
```diff
@@ -38,40 +36,34 @@ export const getConnectedCalendars = async (
...
-            isSelected: selectedCalendars.some((selected) => selected.externalId === cal.externalId),
-            credentialId,
-          }))
-          .sortBy(["primary"])
-          .value();
-        const primary = calendars.find((item) => item.primary) ?? calendars[0];
-        if (!primary) {
-          throw new Error("No primary calendar found");
-        }
+      if (!calendar) {
```
Comment (leog) [inline packages/core/CalendarManager.ts:40]: Replacing the try/catch with the proper short-circuit return value that enables showing the warning without needing to catch an error or even showing it.

File: packages/ui/v2/core/Alert.tsx [hunk 6]
```diff
@@ -7,10 +7,11 @@ export interface AlertProps {
   message?: ReactNode;
   actions?: ReactNode;
   className?: string;
+  iconClassName?: string;
   severity: "success" | "warning" | "error" | "info";
 }
 export function Alert(props: AlertProps) {
-  const { severity } = props;
+  const { severity, iconClassName } = props;
```
Comment (leog) [inline packages/ui/v2/core/Alert.tsx:14]: To be able to style the icon in the alert as I needed for the calendar list

File: packages/ui/v2/core/List.tsx [hunk 7]
```diff
@@ -3,12 +3,14 @@ import { createElement } from "react";
...
+export function List(props: JSX.IntrinsicElements["ul"] & { noBorderTreatment?: true }) {
+  const { noBorderTreatment, ...rest } = props;
   return (
     <ul
-      {...props}
+      {...rest}
       className={classNames(
-        "divide-y divide-neutral-200 rounded-md border border-l border-r sm:mx-0 sm:overflow-hidden",
+        "sm:mx-0 sm:overflow-hidden",
+        !noBorderTreatment && "divide-y divide-neutral-200 rounded-md border border-l border-r ",
```
Comment (leog) [inline packages/ui/v2/core/List.tsx:13]: no border option, thank God!

**Comment by joeauyeung** [comment]
Before merging we should also make sure this displays on the settings calendar page as well.

**Comment by leog** [comment]
Took the opportunity to improve the mobile experience aside from the needed type fix.

From this:
<kbd></kbd>

To this:
<kbd></kbd>

**Comment by hariombalhara** [comment]
@leog Can this be fixed along with it ? https://github.com/calcom/cal.com/issues/3204

**Comment by hariombalhara** [comment]
Turns out it was quick to do. I have made the changes and this is how it should look. @Jaibles for design on showing a link in the alert
Screenshot 2022-09-21 at 6 37 21 PM

I picture the ideal flow should be that when user clicks delete button(which should be changed to a reconnect icon), the app should be uninstalled the same way it is currently and user should be redirected to App installation page to reinstall it(preferably directly to the URL that is behind Install button)
This would make the entire reconnect flow smooth

**Review by hariombalhara — APPROVED** [review]

**Pull request merged**